## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#bbfd5a19-e4e1-4f4e-9e30-ec7178a44470) |
| mint-large-bucket.sh | :x: [more...](#518eefce-6965-46b7-b340-a7c9e44542e4) |
| mint-gateway-nas.sh | :x: [more...](#7654a1c0-1e51-46d8-877c-c897e7425dbb) |
<a name="bbfd5a19-e4e1-4f4e-9e30-ec7178a44470"></a>
### 8574-77d61a9/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:30362
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f41f6ae3387e:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-acdwaqkffnp4bn",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364274,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15DA6543C325CA62, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="7654a1c0-1e51-46d8-877c-c897e7425dbb"></a>
### 8574-77d61a9/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:30128
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 02bf678f19aa:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 1 seconds
(5/14) Running awscli tests ... done in 57 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 28 seconds
(8/14) Running minio-dotnet tests ... done in 21 seconds
(9/14) Running minio-go tests ... done in 26 seconds
(10/14) Running minio-java tests ... done in 15 seconds
(11/14) Running minio-js tests ... done in 40 seconds
(12/14) Running minio-py tests ... done in 57 seconds
(13/14) Running s3cmd tests ... FAILED in 6 seconds
{
  "name": "s3cmd",
  "duration": "2077",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  following lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\nInvoked as: /usr/local/bin/s3cmd --config /tmp/.s3cmd-4871/s3cfg get s3://s3cmd-test-bucket-5038/s3cmd-test-object-1945 s3cmd-test-object-1945.downloaded\nProblem: <class 'ConnectionResetError: [Errno 104] Connection reset by peer\nS3cmd:   2.0.2\npython:   3.5.2 (default, Oct  8 2019, 13:06:37) \n[GCC 5.4.0 20160609]\nenvironment LANG=C.UTF-8\n\nTraceback (most recent call last):\n  File \"/usr/local/bin/s3cmd\", line 3092, in <module>\n    rc = main()\n  File \"/usr/local/bin/s3cmd\", line 3001, in main\n    rc = cmd_func(args)\n  File \"/usr/local/bin/s3cmd\", line 590, in cmd_object_get\n    response = s3.object_get(uri, dst_stream, destination, start_position = start_position, extra_label = seq_label)\n  File \"/usr/local/lib/python3.5/dist-packages/S3/S3.py\", line 716, in object_get\n    response = self.recv_file(request, stream, labels, start_position)\n  File \"/usr/local/lib/python3.5/dist-packages/S3/S3.py\", line 1742, in recv_file\n    data = http_response.read(this_chunk)\n  File \"/usr/lib/python3.5/http/client.py\", line 458, in read\n    n = self.readinto(b)\n  File \"/usr/lib/python3.5/http/client.py\", line 498, in readinto\n    n = self.fp.readinto(b)\n  File \"/usr/lib/python3.5/socket.py\", line 575, in readinto\n    return self._sock.recv_into(b)\nConnectionResetError: [Errno 104] Connection reset by peer\n\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n    An unexpected error has occurred.\n  Please try reproducing the error using\n  the latest s3cmd code from the git master\n  branch found at:\n    https://github.com/s3tools/s3cmd\n  and have a look at the known issues list:\n    https://github.com/s3tools/s3cmd/wiki/Common-known-issues-and-their-solutions\n  If the error persists, please report the\n  above lines (removing any private\n  info as necessary) to:\n   s3tools-bugs@lists.sourceforge.net\n!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
}

Executed 12 out of 14 tests successfully.
```
<a name="518eefce-6965-46b7-b340-a7c9e44542e4"></a>
### 8574-77d61a9/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:31273
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e0dd9dc7eb74:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 52 seconds
(4/14) Running aws-sdk-ruby tests ... done in 6 seconds
(5/14) Running awscli tests ... done in 2 minutes and 25 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 1 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 5 seconds
(9/14) Running minio-go tests ... FAILED in 3 minutes and 54 seconds
{
  "args": {},
  "duration": 76585,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8574-77d61a9

--- a/cmd/http/server.go
+++ b/cmd/http/server.go
@@ -21,7 +21,6 @@ import (
 	"errors"
 	"io/ioutil"
 	"net/http"
-	"os"
 	"runtime/pprof"
 	"sync"
 	"sync/atomic"
@@ -143,7 +142,7 @@ func (srv *Server) Shutdown() error {
 			// Write all running goroutines.
 			tmp, err := ioutil.TempFile("", "minio-goroutines-*.txt")
 			if err == nil {
-				_ = pprof.Lookup("goroutine").WriteTo(os.Stderr, 1)
+				_ = pprof.Lookup("goroutine").WriteTo(tmp, 1)
 				tmp.Close()
 				return errors.New("timed out. some connections are still active. doing abnormal shutdown. goroutines written to %s" + tmp.Name())
 			}

--- a/cmd/http/server.go
+++ b/cmd/http/server.go
@@ -19,6 +19,7 @@ package http
 import (
 	"crypto/tls"
 	"errors"
+	"io/ioutil"
 	"net/http"
 	"os"
 	"runtime/pprof"
@@ -139,8 +140,13 @@ func (srv *Server) Shutdown() error {
 	for {
 		select {
 		case <-shutdownTimer.C:
-			// Print all running goroutines.
-			_ = pprof.Lookup("goroutine").WriteTo(os.Stderr, 1)
+			// Write all running goroutines.
+			tmp, err := ioutil.TempFile("", "minio-goroutines-*.txt")
+			if err == nil {
+				_ = pprof.Lookup("goroutine").WriteTo(os.Stderr, 1)
+				tmp.Close()
+				return errors.New("timed out. some connections are still active. doing abnormal shutdown. goroutines written to %s" + tmp.Name())
+			}
 			return errors.New("timed out. some connections are still active. doing abnormal shutdown")
 		case <-ticker.C:
 			if atomic.LoadInt32(&srv.requestCount) <= 0 {

--- a/cmd/http/server.go
+++ b/cmd/http/server.go
@@ -144,7 +144,7 @@ func (srv *Server) Shutdown() error {
 			if err == nil {
 				_ = pprof.Lookup("goroutine").WriteTo(tmp, 1)
 				tmp.Close()
-				return errors.New("timed out. some connections are still active. doing abnormal shutdown. goroutines written to %s" + tmp.Name())
+				return errors.New("timed out. some connections are still active. doing abnormal shutdown. goroutines written to " + tmp.Name())
 			}
 			return errors.New("timed out. some connections are still active. doing abnormal shutdown")
 		case <-ticker.C:

